src/python/pants/bin/pants_runner.py

```diff
@@ -42,10 +42,18 @@ def _enable_rust_logging(self, global_bootstrap_options):
     setup_logging_to_stderr(logging.getLogger(None), levelname)
 
   def _should_run_with_pantsd(self, global_bootstrap_options):
+    # When running with pantsd if a ./pants command, somewhere in its process calls ./pants again
+    # (e.g. to build a pex), the inner run will wait for the outer run to finish,
+    # which will never happen.
+    # The parent_build_id option is set only for pants inner runs which parent is pantsd run.
+    # This will allow us not to run them with pantsd and to know who was the parent
+    # of pants inner runs.
+    is_inner_run_from_pantsd = global_bootstrap_options.parent_build_id is not None
```

Comment:
My reading of how this option is set, it's not actually specific to whether the parent is a pantsd or just a pants (non-pantsd runs will also use `LocalPantsRunner`), so maybe remove `pantsd` from this variable name?

Reply:
I was thinking about it. 

If the 'parent_build_id' flag is used to not use pantsd when a pants run has a parent than:
 if the parent doesn't use pantsd, than children also won't be able even if they need to (e.g. tests that test pantsd with inner runs). 

For now, the flag is set only if a parent used pantsd:
`if self._is_daemon:
      os.environ['PANTS_PARENT_BUILD_ID'] = self._run_tracker.run_id`

Reply:
> If the 'parent_build_id' flag is used to not use pantsd when a pants run has a parent than
That happens to be how we're setting it today, but it's not what the flag means. If we really want to restrict this to pantsd-only, we should mention pantsd in the name of the `--parent-build-id` flag. I don't think we want to though, because this is a general concept which applies regardless of pantstd.